aws-iot-securityprofile/aws-iot-securityprofile.json

```diff
@@ -0,0 +1,345 @@
+      "properties": {
+        "Statistic": {
+          "description": "The percentile which resolves to a threshold value by which compliance with a behavior is determined",
+          "type": "string",
+          "enum": [
+            "Average",
+            "p0",
+            "p0.1",
+            "p0.01",
+            "p1",
+            "p10",
+            "p50",
+            "p90",
+            "p99",
+            "p99.9",
+            "p99.99",
+            "p100"
+          ]
+        }
+      },
+      "additionalProperties": false
+    },
+    "AlertTarget": {
+      "description": "A structure containing the alert target ARN and the role ARN.",
+      "type": "object",
+      "properties": {
+        "AlertTargetArn": {
+          "description": "The ARN of the notification target to which alerts are sent.",
+          "type": "string",
+          "maxLength": 2048
+        },
+        "RoleArn": {
+          "description": "The ARN of the role that grants permission to send alerts to the notification target.",
+          "type": "string",
+          "minLength": 20,
+          "maxLength": 2048
+        }
+      },
+      "required": [
+        "AlertTargetArn",
+        "RoleArn"
+      ],
+      "additionalProperties": false
+    },
+    "MetricToRetain": {
+      "description": "The metric you want to retain. Dimensions are optional.",
+      "type": "object",
+      "properties": {
+        "Metric": {
+          "description": "What is measured by the behavior.",
+          "type": "string",
+          "pattern": "[a-zA-Z0-9:_-]+",
+          "minLength": 1,
+          "maxLength": 128
+        },
+        "MetricDimension": {
+          "$ref": "#/definitions/MetricDimension"
+        }
+      },
+      "required": [
+        "Metric"
+      ],
+      "additionalProperties": false
+    },
+    "Tag": {
+      "description": "A key-value pair to associate with a resource.",
+      "type": "object",
+      "properties": {
+        "Key": {
+          "type": "string",
+          "description": "The tag's key.",
+          "minLength": 1,
+          "maxLength": 128
+        },
+        "Value": {
+          "type": "string",
+          "description": "The tag's value.",
+          "minLength": 1,
+          "maxLength": 256
+        }
+      },
+      "required": [
+        "Value",
+        "Key"
+      ],
+      "additionalProperties": false
+    }
+  },
+  "properties": {
+    "SecurityProfileName": {
+      "description": "A unique identifier for the security profile.",
+      "type": "string",
+      "pattern": "[a-zA-Z0-9:_-]+",
+      "minLength": 1,
+      "maxLength": 128
+    },
+    "SecurityProfileDescription": {
+      "description": "A description of the security profile.",
+      "type": "string",
+      "maxLength": 1000
+    },
+    "Behaviors": {
+      "description": "Specifies the behaviors that, when violated by a device (thing), cause an alert.",
+      "type": "array",
+      "maxLength": 100,
```

Comment:
don't think `minLength`/`maxLength` can be applied to `array` types:
https://github.com/aws-cloudformation/cloudformation-cli/issues/414
https://json-schema.org/understanding-json-schema/reference/array.html#length